src/main/webapp/app/shared/components/code-button/code-button.component.html

```diff
@@ -20,44 +21,41 @@
         @if (useSsh && areAnySshKeysExpired) {
             <div class="alert alert-warning" [innerHTML]="sshKeysExpiredTip"></div>
         }
-        @if (useToken && tokenMissing) {
+        @if (useToken && isInCourseManagement && !userTokenPresent) {
             <div class="alert alert-warning" [innerHTML]="tokenMissingTip"></div>
         }
-        @if (useToken && tokenExpired) {
+        @if (useToken && isInCourseManagement && userTokenPresent && !userTokenStillValid) {
             <div class="alert alert-warning" [innerHTML]="tokenExpiredTip"></div>
         }
-        @if (participations && participations.length > 1) {
+        @if (participations()!.length > 1) {
             <div class="form-check form-switch">
                 <input mdbCheckbox id="switch-practice" class="form-check-input" type="checkbox" [checked]="isPracticeMode" (click)="switchPracticeMode()" />
                 <label class="form-check-label" for="switch-practice" jhiTranslate="artemisApp.exerciseActions.practiceMode.title"></label>
             </div>
         }
-        <h5>{{ cloneHeadline | artemisTranslate }}</h5>
+        <h5 [jhiTranslate]="cloneHeadline"></h5>
+
         <div class="d-flex" style="margin: 0; max-width: 100%">
-            @if (sshEnabled) {
-                <div aria-label="Button group with nested dropdown" class="btn-group" ngbDropdown role="group" style="margin-bottom: 10px">
-                    <button class="btn btn-primary btn-sm dropdown-toggle https-or-ssh-button" ngbDropdownToggle type="button">
-                        {{ useSsh ? 'SSH' : useToken ? 'Token' : 'HTTPS' }}
-                    </button>
-                    <div class="dropdown-menu" ngbDropdownMenu style="min-width: 100%">
-                        @if (showCloneUrlWithoutToken) {
-                            <a (click)="useHttpsUrlWithoutToken()" class="dropdown-item {{ !useToken && !useSsh ? 'active' : '' }}" id="useHTTPSButton">HTTPS</a>
+            <div aria-label="Button group with nested dropdown" class="btn-group" ngbDropdown role="group" style="margin-bottom: 10px">
+                <button class="btn btn-primary btn-sm dropdown-toggle https-or-ssh-button" ngbDropdownToggle type="button">
+                    {{ useSsh ? 'SSH' : useToken ? 'Token' : 'HTTPS' }}
+                </button>
+                <div class="dropdown-menu" ngbDropdownMenu style="min-width: 100%">
+                    @for (method of authenticationMethods; track method) {
+                        @if (method === 'password') {
+                            <a (click)="useHttpsPassword()" class="dropdown-item {{ !useToken && !useSsh ? 'active' : '' }}" id="useHTTPSButton">HTTPS</a>
```

Comment:
we could think of moving this to a separate function as this seems somewhat error-prone to me. sth like `isAuthMethodActivated(method: SSHAuthMethod): boolean` might make it better understandable

Reply:
Not sure if I get how you mean that tbh. The idea behind this is, that depending on the order of elements in `authenticationMethods` the options are shown. E.g.: 
![image](https://github.com/user-attachments/assets/7ba9857f-67cc-4692-9be1-a6e1107c2e1f)
 or 
![image](https://github.com/user-attachments/assets/caf87cc1-1eeb-4aad-8858-869421b4efc5)
or
![image](https://github.com/user-attachments/assets/069b6883-ffa7-42af-a94c-067582b9e6e5)
depending on the configuration of the server